packages/core/http/core-http-server-internal/src/http_server.ts

```diff
@@ -55,6 +57,51 @@ import { AuthHeadersStorage } from './auth_headers_storage';
 import { BasePath } from './base_path_service';
 import { getEcsResponseLog } from './logging';
 
+const THRESHOLD_ELU = 0.15;
+const THRESHOLD_ELA = 250;
+
+/**
+ * Adds ELU timings for the executed function to the current's context transaction
+ *
+ * @param path The request path
+ * @param log  Logger
+ */
+function startEluMeasurement<T>(path: string, log: Logger): () => void {
+  const startUtilization = performance.eventLoopUtilization();
+  const start = performance.now();
+
+  return function stopEluMeasurement() {
+    const { active, utilization } = performance.eventLoopUtilization(startUtilization);
+
+    apm.currentTransaction?.addLabels(
+      {
+        event_loop_utilization: utilization,
+        event_loop_active: active,
+      },
+      false
+    );
+
+    const duration = performance.now() - start;
+
+    if (active > THRESHOLD_ELA && utilization > THRESHOLD_ELU) {
+      log.warn(
```

Comment:
I'm concerned that on an already overloaded server this will constantly trigger increasing the load even further because of the load that logging adds. We can run the performance journeys to confirm that after a certain load we start logging for each API request.

But unless I'm wrong, I think we should only log in development or have some kind of rate-limiting.

Reply:
Agree, I'd like to turn it off by default and make it opt-in. Do you have an idea of what the config setting should be and how I get it into http_server?